src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_imad.cl

```diff
@@ -36,19 +37,30 @@ KERNEL(fully_connected_gpu_imad)(
 #if HAS_OFM_LEFTOVERS || HAS_IFM_LEFTOVERS
     const uint sglid = get_sub_group_local_id();
 #endif
+#if IS_DYNAMIC
+    // In dynamic kernel, TILE_BATCH is set to the initial tile batch size for stack arrays such as dotProd
+    // and tile_batch is calculated as an adjusted value from tile_batch_max_size by given global work size
+#if OUTPUT_3D
+    const uint tile_batch = OUTPUT_FEATURE_NUM / (uint)get_global_size(2);
+#else
+    const uint tile_batch = OUTPUT_BATCH_NUM / (uint)get_global_size(1);
+#endif
+#else
+    const uint tile_batch = TILE_BATCH;
+#endif
 
 #if OUTPUT_3D
     const uint batch = (uint)get_global_id(1);
-    const uint skip_f = (uint)get_global_id(2) * TILE_BATCH;
+    const uint skip_f = (uint)get_global_id(2) * tile_batch;
 #else
-    const uint batch = (uint)get_global_id(1) * TILE_BATCH;
+    const uint batch = (uint)get_global_id(1) * tile_batch;
     const uint skip_f = (uint)get_global_id(2);
 #endif
 
     // Accumulators initialization
     MAKE_VECTOR_TYPE(int, TILE_OFM) dotProd[TILE_BATCH];
     MAKE_VECTOR_TYPE(uint, TILE_OFM) idx_w;
-    unroll_for (uint ob_idx = 0; ob_idx < TILE_BATCH; ob_idx++) {
+    unroll_for (uint ob_idx = 0; ob_idx < tile_batch; ob_idx++) {
```

Comment:
For shape agnostic kernel, tile_batch is not constant, so please do not "unroll_for".

Reply:
update to do not loop unrolling for shape agnostic kernel